src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/SVFileUtils.java

```diff
@@ -0,0 +1,176 @@
+package org.broadinstitute.hellbender.tools.spark.sv.utils;
+
+import htsjdk.samtools.*;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.utils.HopscotchSet;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+
+import java.io.*;
+import java.util.*;
+
+public final class SVFileUtils {
+
+    private static final String REFERENCE_GAP_INTERVAL_FILE_COMMENT_LINE_PROMPT = "#";
+
+    /**
+     * Creates a directory, in local FS, HDFS, or Google buckets to write individual files in.
+     */
+    public static void createDirectory(final String pathString) throws IOException {
+        Utils.nonNull(pathString);
+
+        if (BucketUtils.isHadoopUrl(pathString)) {
```

Comment:
I still cannot understand why a Hadoop path should be treated different than another kind of path. I believe that the `IOUtils.getPath` will get properly the Hadoop path, and then `Files.createDirectory` should get the `HadoopFileSystemProvider` for the HDFS, which in turn uses the mkdirs that you are using here.

Sorry if I am missing something....

Reply:
done what's suggested here, moved to IOUtils and tested

Reply:
@magicDGS, in theory you are right but in practice there seems to be some issues (see https://github.com/broadinstitute/gatk/issues/3569 https://github.com/broadinstitute/barclay/issues/106)
Perhaps now we could drop the handoop but as far as this PR is concern it should follow the current approach.

Reply:
Actually, the problem is mainly with the GCS provider as far as I know (If I am wrong, maybe @lbergelson or @droazen can chime in for correct me).

The discussion in barclay is pointing to the method in `IOUtils`, as the hack for getting correctly the paths, which is working perfectly fine for hadoop (I have been using it in my own code properly).

In the case of this code, if the path is get by `IOUtils.getPath` the providers are loaded on demand (including the special case of GCS, which is handled with `BucketUtils`). Thus, using `Files` is possible once the providers are loaded. Again, as far as I know, the __jsr203-hadoop__ provider is working perfectly fine with `Files` (tested in my own toolkit) and that is the only special case handled here. I still do not fully understand why that's the case, while the GCS code is handled by `Files` in the implementation pointed by your comment. In principle, I think that most of the problems with providers are found with the GCS loading code...but again, @lbergelson or @droazen can correct me if I am missing something...